crates/rolldown/tests/esbuild/default/assign_to_import/artifacts.snap

```diff
@@ -2,110 +2,113 @@
+   ╭─[bad0.js:1:22]
+   │
+ 1 │ import x from "foo"; x = 1
+   │                      ┬  
+   │                      ╰── Imports are immutable in JavaScript. To modify the value of this import, you must export a setter function in the imported file (e.g. 'setX') and then import and call that function here instead.
+───╯
+
+```
+## ASSIGN_TO_IMPORT
+
+```text
+[ASSIGN_TO_IMPORT] Error: Cannot assign to import 'x'
+   ╭─[bad1.js:1:22]
+   │
+ 1 │ import x from "foo"; x++
+   │                      ┬  
+   │                      ╰── Imports are immutable in JavaScript. To modify the value of this import, you must export a setter function in the imported file (e.g. 'setX') and then import and call that function here instead.
+───╯
+
+```
+## ASSIGN_TO_IMPORT
+
+```text
+[ASSIGN_TO_IMPORT] Error: Cannot assign to import 'x'
+   ╭─[bad2.js:1:24]
+   │
+ 1 │ import x from "foo"; ([x] = 1)
+   │                        ┬  
+   │                        ╰── Imports are immutable in JavaScript. To modify the value of this import, you must export a setter function in the imported file (e.g. 'setX') and then import and call that function here instead.
+───╯
+
+```
+## ASSIGN_TO_IMPORT
+
+```text
+[ASSIGN_TO_IMPORT] Error: Cannot assign to import 'x'
+   ╭─[bad3.js:1:24]
+   │
+ 1 │ import x from "foo"; ({x} = 1)
+   │                        ┬  
+   │                        ╰── Imports are immutable in JavaScript. To modify the value of this import, you must export a setter function in the imported file (e.g. 'setX') and then import and call that function here instead.
+───╯
+
+```
+## ASSIGN_TO_IMPORT
+
+```text
+[ASSIGN_TO_IMPORT] Error: Cannot assign to import 'x'
+   ╭─[bad4.js:1:27]
+   │
+ 1 │ import x from "foo"; ({y: x} = 1)
+   │                           ┬  
+   │                           ╰── Imports are immutable in JavaScript. To modify the value of this import, you must export a setter function in the imported file (e.g. 'setX') and then import and call that function here instead.
+───╯
+
+```
+## ASSIGN_TO_IMPORT
+
+```text
+[ASSIGN_TO_IMPORT] Error: Cannot assign to import 'x'
+   ╭─[bad5.js:1:24]
+   │
+ 1 │ import {x} from "foo"; x++
+   │                        ┬  
+   │                        ╰── Imports are immutable in JavaScript. To modify the value of this import, you must export a setter function in the imported file (e.g. 'setX') and then import and call that function here instead.
+───╯
+
+```
+## ASSIGN_TO_IMPORT
+
+```text
+[ASSIGN_TO_IMPORT] Error: Cannot assign to import 'x'
+   ╭─[bad6.js:1:27]
+   │
+ 1 │ import * as x from "foo"; x++
+   │                           ┬  
+   │                           ╰── Imports are immutable in JavaScript. To modify the value of this import, you must export a setter function in the imported file (e.g. 'setX') and then import and call that function here instead.
+───╯
+
+```
+## ASSIGN_TO_IMPORT
+
+```text
+[ASSIGN_TO_IMPORT] Error: Cannot assign to import 'x'
+   ╭─[good0.js:1:28]
```

Comment:
Maybe false positive case in oxc, https://oxc-project.github.io/oxc/playground/?code=3YCAAICigICAgICAgIC0G8rnONK89ITJ3zrK%2FUP5L%2FiZLeiyawJpf0w9TqWLqZyk6ZSj338lRQCA